homeassistant/components/mill/number.py

```diff
@@ -0,0 +1,82 @@
+"""Support for mill wifi-enabled home heaters."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from homeassistant.components.number import NumberDeviceClass, NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_USERNAME, UnitOfPower
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CLOUD, CONNECTION_TYPE, DOMAIN, MANUFACTURER
+
+if TYPE_CHECKING:
+    from .coordinator import MillDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Mill Number."""
+    if entry.data.get(CONNECTION_TYPE) == CLOUD:
+        mill_data_coordinator: MillDataUpdateCoordinator = hass.data[DOMAIN][CLOUD][
+            entry.data[CONF_USERNAME]
+        ]
+
+        async_add_entities(
+            MillNumber(mill_data_coordinator, mill_device)
+            for mill_device in mill_data_coordinator.data.values()
+        )
+
+
+class MillNumber(CoordinatorEntity, NumberEntity):
+    """Representation of a Mill number device."""
+
+    _attr_device_class = NumberDeviceClass.POWER
+    _attr_has_entity_name = True
+    _attr_native_max_value = 2000
+    _attr_native_min_value = 0
+    _attr_native_step = 1
+    _attr_native_unit_of_measurement = UnitOfPower.WATT
+
+    def __init__(self, coordinator: MillDataUpdateCoordinator, mill_device) -> None:
+        """Initialize the number."""
+        super().__init__(coordinator)
+
+        self.coordinator: MillDataUpdateCoordinator = coordinator
```

Comment:
```suggestion
```
Should already be set by `CoordinatorEntity`

Reply:
Yes, but then I got problems with mypy
Do you have any suggestions on how to fix that? Or should I ignore that line?

Reply:
https://github.com/home-assistant/core/actions/runs/12506870702/job/34892521086?pr=134044#step:9:28

Reply:
`class MillNumber(CoordinatorEntity[MillDataUpdateCoordinator], NumberEntity):`

Reply:
Thanks 👍